### PR TITLE
[ENH] Added Flag to save `LogFile`

### LINKED ISSUE #250: Add flag for `add_data_to_graph.sh` to write output to a log file
If you want to be able to check the detailed HTTP response for files that failed the upload (summarized at end of script output), currently you either have to scroll up in your terminal through the whole script STOUT, or make sure to direct the output of `add_data_to_graph.sh` to a file as you're running it.

Having a flag for the script to write all output to a LOG.txt or similar file could be very useful to simplify debugging.

### PR DESCRIPTION
**Pull request by samadpls**
- Closes #250 

Changes proposed in this pull request:

-  Added flag to save log 

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [ ] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #250 `
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

**Review by alyssadai — CHANGES_REQUESTED** [review]
Thanks a lot for the PR @samadpls! 🎉

This is looking good, though I think a few more changes are needed for full functionality of your new `--log-output` flag. See my comments/suggestions below, and feel free to let me know if you have any questions!

Please re-request a review once you have addressed the feedback.

**Comment by samadpls** [comment]
Hi @alyssadai,
Thank you for your review 😋. I have implemented the suggested changes and addressed all the comments. 
Thanks again!

File: add_data_to_graph.sh [hunk 1]
```diff
@@ -145,71 +166,81 @@ else
...
-echo "FINISHED UPLOADING DATA FROM ${jsonld_dir} TO ${graph_db}."
-
-if (( ${#upload_failed[@]} != 0 )); then
-	echo -e "\nERROR: Upload failed for these files:"
-	printf '%s\n' "${upload_failed[@]}"
+# Call the main logic function with or without output redirection
+if [ "$log_output" = "on" ]; then
+    if [ -z "$log_file" ]; then
+        log_file="LOG.txt"
+    fi
```
Comment (alyssadai) [inline add_data_to_graph.sh:241]:
```suggestion
```

This `if` block shouldn't be necessary: the logic for the default log file and empty arg handling is handled by the argument parsing code in the argbash block above.

File: add_data_to_graph.sh [hunk 2]
```diff
@@ -145,71 +166,81 @@ else
...
+    main >> "$log_file"
```
Comment (alyssadai) [inline add_data_to_graph.sh:242]:
```suggestion
    main > "$log_file"
```
for easier separation of logs from different runs

**Review by alyssadai — COMMENTED** [review]
Hey @samadpls, thanks for the changes!

Two final suggestions below, then I think this is good to be merged 🧑‍🍳

**Comment by alyssadai** [comment]
Looks great, thanks a lot for your contribution @samadpls!

**Comment by alyssadai** [comment]
Oops, accidentally closed instead of merged.

**Review by alyssadai — APPROVED** [review]

**Pull request merged**